OpenRA.Mods.Common/Activities/Enter.cs

```diff
@@ -74,6 +80,11 @@ public override Activity Tick(Actor self)
 			var oldUseLastVisibleTarget = useLastVisibleTarget;
 			useLastVisibleTarget = targetIsHiddenActor || !target.IsValidFor(self);
 
+			// Cancel if the target changed owner, and the valid owner check fails.
+			bool ownerHasChanged = target.GetTargetType(true) != target.GetTargetType(false);
```

Comment:
Comparing the owner generations should give us this information too, right?

Also, please use `var` instead of the explicit type `bool`.

Reply:
I'm new to C# so I'm not sure why we don't explicitly type things here when we have the option to do so. In this case, It seems like bool is more concise and clear. 
I can use var here if you guys would prefer. 

And ownerGeneration / generation are private properties, and would not be visible from here.

Reply:
> And ownerGeneration / generation are private properties, and would not be visible from here.

Gotcha.  I my mind was on another changeset I saw a few days earlier.

> I'm not sure why we don't explicitly type things here when we have the option to do so. In this case, It seems like bool is more concise and clear. This could just be that I'm new to c#.

It is OpenRA's convention to use implicit typing (via `var`) when the local's type doesn't *need* to be explicitly declared.

An example of needing the type to be declared is, for some reason, needing the base type or forcing the result to be an interface instead of a concrete type.